### PR TITLE
Fix paths for post action 'Add project to a solution file'

### PR DESCRIPTION
**Pull request by AR-May**
Delivers issue #1991 (precisely [this comment](https://github.com/dotnet/templating/issues/1991#issuecomment-686299488)). 

The post-action "Add projects to a solution file" sometimes fails if the `-o` or `-n` option presents. The changes fix paths and add unit tests for non-empty `-o` option.

### CODE REVIEW

**Review by ladipro — COMMENTED** [review]
Can you please confirm that the solution file ends up with relative paths wherever possible, just like before this change? I.e. this is fixing the arguments passed to `dotnet sln add` but the actual edits done to the sln file are the same. Thank you!

**Comment by AR-May** [comment]
> Can you please confirm that the solution file ends up with relative paths wherever possible, just like before this change? I.e. this is fixing the arguments passed to `dotnet sln add` but the actual edits done to the sln file are the same. Thank you!

Yes, I confirm that the resulted solution file ends up with paths that are relative to the solution file location. The `dotnet sln add` takes care of that.
Well, moreover, if you look a few lines above my changes (line 82), you will see that if `primaryOutputIndexes` present (that means we apply post action to the subset of primary outputs, not to the whole set), there is the same transformation of the path. So, I see this as a correct way to fix the bug.

**Review by ladipro — APPROVED** [review]

File: test/Microsoft.TemplateEngine.Cli.UnitTests/AddProjectsToSolutionPostActionTests.cs [hunk 1]
```diff
@@ -101,5 +101,68 @@ public void AddProjectToSolutionPostActionDoesntFindProjectOutOfRange()
             Assert.False(AddProjectsToSolutionPostAction.TryGetProjectFilesToAdd(EngineEnvironmentSettings, postAction, creationResult, string.Empty, out IReadOnlyList<string> foundProjectFiles));
             Assert.Null(foundProjectFiles);
         }
+
+        [Fact(DisplayName = nameof(AddProjectToSolutionPostActionFindsMultipleProjectsToAddWithOutputBasePath))]
```
[the remark this answers is not in the document]
Comment (AR-May) [reply test/Microsoft.TemplateEngine.Cli.UnitTests/AddProjectsToSolutionPostActionTests.cs:105]: Apparently, there is a difference: the default display name contains also the namespace of the test "Microsoft.TemplateEngine.Cli.UnitTests.".

[the remark this answers is not in the document]
Comment (ladipro) [reply test/Microsoft.TemplateEngine.Cli.UnitTests/AddProjectsToSolutionPostActionTests.cs:105]: Interesting, thanks. I wonder where the fully qualified name would cause issues. Or where it would surface at all, since Test Explorer is evidently showing bare method names even without `DisplayName`:

image

Anyway, just a random comment, feel free to ignore. Thanks!
